nakadi/hack.py

```diff
@@ -120,71 +122,44 @@ def __get_partitions_offsets(topic):
-    if not valid:
-        return result
-    stream_opts['batch_flush_timeout'] = result
-
-    valid, result = __try_get_parameter_as_int('batch_keep_alive_limit', flask.request, False, -1)
-    if not valid:
-        return result
-    stream_opts['batch_keep_alive_limit'] = result
-
-    valid, result = __try_get_parameter_as_int('stream_limit', flask.request, False, 0)
-    if not valid:
-        return result
-    stream_opts['stream_limit'] = result
-
-    valid, result = __try_get_parameter_as_int('stream_timeout', flask.request, False, 0)
-    if not valid:
-        return result
-    stream_opts['stream_timeout'] = result
-
-    # check that partition is integer
-    if not partition.isdigit():
-        return {'detail': '"partition" path parameter should be an integer number'}, 400
-    else:
-        partition = int(partition)
-
-    # check that topic and partition exist
+    try:
+        stream_opts['batch_limit'] = get_int_parameter('batch_limit', flask.request, False, 1)
+        stream_opts['batch_flush_timeout'] = get_int_parameter('batch_flush_timeout', flask.request, False, 0)
+        stream_opts['batch_keep_alive_limit'] = get_int_parameter('batch_keep_alive_limit', flask.request, False, -1)
+        stream_opts['stream_limit'] = get_int_parameter('stream_limit', flask.request, False, 0)
+        stream_opts['stream_timeout'] = get_int_parameter('stream_timeout', flask.request, False, 0)
+    except NotIntegerParameterException as e:
+        return {'detail': '"%s" query parameter should be an integer number' % e.parameter}, 400
+    except RequiredParameterNotFoundException as e:
+        return {'detail': 'missing required query parameter "%s"' % e.parameter}, 400
+
+    # check that topic exists
     if not __topic_exists(topic):
         return {'detail': 'topic not found'}, 404
-    if not __partition_exists(topic, partition):
-        return {'detail': 'partition not found'}, 404
+
+    # get cursors to start reading from
+    cursors_str = flask.request.headers.get('x-nakadi-cursors')
+    if not cursors_str:
+        # if cursors are not specified - read from all partitions from the latest offset
+        partitions_offsets = __get_partitions_offsets(topic)
+        cursors = [{
+                       'partition': offset['partition_id'],
+                       'offset': offset['newest_available_offset']
+                   } for offset in partitions_offsets]
+    else:
+        try:
+            cursors = get_cursors(cursors_str)
+        except WrongCursorsFormatException:
+            return {'detail': '"x-nakadi-cursors" header has wrong format'}, 400
 
     # returning generator in response will create a stream
-    stream_generator = event_stream.create_stream_generator(kafka_pool, topic, partition, start_from, stream_opts)
+    stream_generator = event_stream.create_stream_generator(kafka_pool, topic, cursors, stream_opts)
     return flask.Response(stream_generator, mimetype = 'text/plain', status = 200)
```

Comment:
@valgog Well, it's definitely not json (as we have \n separators), so for now text/plain would be fine I think